op-devstack/sysgo/l2_proposer_faultproof.go

```diff
@@ -0,0 +1,332 @@
+	orch.proposerOptions.Apply(p, proposerID, cfg)
+	for _, opt := range opts {
+		opt(p, proposerID, cfg)
+	}
+
+	require.NotEmpty(cfg.l1ConfigDir, "fault-proof proposer L1 config dir must be set")
+	require.NotEmpty(cfg.l2ConfigDir, "fault-proof proposer L2 config dir must be set")
+
+	l1RPC := l1EL.UserRPC()
+	l1BeaconRPC := l1CL.beaconHTTPAddr
+	l2RPC := strings.ReplaceAll(l2EL.UserRPC(), "ws://", "http://")
+	l2NodeRPC := strings.ReplaceAll(l2CL.UserRPC(), "ws://", "http://")
+	mockVerifierAddr := l2Net.deployment.sp1MockVerifier
+	disputeGameFactoryProxy := l2Net.deployment.disputeGameFactoryProxy
+
+	logger.Info("L1_RPC", "url", l1RPC)
+	logger.Info("L1_BEACON_RPC", "url", l1BeaconRPC)
+	logger.Info("L2_RPC", "url", l2RPC)
+	logger.Info("L2_NODE_RPC", "url", l2NodeRPC)
+	logger.Info("SP1MockVerifier", "address", mockVerifierAddr)
+	logger.Info("DisputeGameFactory", "address", disputeGameFactoryProxy)
+
+	proposalIntervalInBlocks := cfg.resolveFdgProposalIntervalInBlocks()
+	fetchInterval := cfg.resolveFdgFetchIntervalInBlocks()
+	fastFinalityMode := cfg.resolveFdgFastFinalityMode()
+	rustLog := cfg.resolveFdgRustLog()
+
+	envVars := map[string]string{
+		"L1_RPC":                      l1RPC,
+		"L1_BEACON_RPC":               l1BeaconRPC,
+		"L2_RPC":                      l2RPC,
+		"L2_NODE_RPC":                 l2NodeRPC,
+		"VERIFIER_ADDRESS":            mockVerifierAddr.String(),
+		"FACTORY_ADDRESS":             disputeGameFactoryProxy.String(),
+		"GAME_TYPE":                   "42",
+		"PRIVATE_KEY":                 proposerKeyStr,
+		"PROPOSAL_INTERVAL_IN_BLOCKS": fmt.Sprintf("%d", proposalIntervalInBlocks),
+		"FETCH_INTERVAL":              fmt.Sprintf("%d", fetchInterval),
+		"FAST_FINALITY_MODE":          fmt.Sprintf("%t", fastFinalityMode),
+		"MOCK_MODE":                   "true",
+		"L1_CONFIG_DIR":               cfg.l1ConfigDir,
+		"L2_CONFIG_DIR":               cfg.l2ConfigDir,
+		"RUST_LOG":                    rustLog,
+		"LOG_FORMAT":                  "json",
+	}
+
+	setEnvFromEnvOrDefault(envVars, "NETWORK_PRIVATE_KEY", "")
+
+	if areMetricsEnabled() {
+		metricsPort, err := getAvailableLocalPort()
+		p.Require().NoError(err, "must get available port for metrics")
+		setEnvFromEnvOrDefault(envVars, "FAULT_PROOF_PROPOSER_METRICS_PORT", metricsPort)
+		envVars["FAULT_PROOF_PROPOSER_METRICS_ENABLED"] = "true"
+	}
+
+	envDir := p.TempDir()
+	envFile := filepath.Join(envDir, fmt.Sprintf("fault-proof-proposer-%s.env", proposerID.String()))
+	err = writeEnvFile(envFile, envVars)
+	p.Require().NoError(err, "must write fault proof proposer env file")
+
+	execPath := os.Getenv("FAULT_PROOF_PROPOSER_EXEC_PATH")
+	p.Require().NotEmpty(execPath, "FAULT_PROOF_PROPOSER_EXEC_PATH environment variable must be set")
+	_, err = os.Stat(execPath)
+	p.Require().NotErrorIs(err, os.ErrNotExist, "executable must exist")
+
+	k := &L2SuccinctFaultProofProposer{
+		id:                 proposerID,
+		userRPC:            "", // retrieved from logs
```

Comment:
Can we just get rid of this field?

Reply:
To avoid touching the `Orchestrator` definition, I'm reusing the same `ProposerID` type for succinct proposers as well. When creating one with `NewL2Proposer`, it technically expects a userRPC, so I'm leaving that empty for now to keep things consistent.